### PR TITLE
[#1990] Move TimeUtil ParseException throwing to ArgsParser::parse method

### PR DESCRIPTION
**Pull request by nseah21**
Fixes [#1990](https://github.com/reposense/RepoSense/issues/1990)
### Proposed commit message

```
ArgsParser::parse uses the methods verifyDatesRangeIsCorrect
and verifySinceDateIsValid from TimeUtil to throw a 
ParseException if either of the checks fail. The throwing 
of an exception in a utility method is a side effect
that makes the methods less reusable.

Let's move the throwing of ParseException from TimeUtil 
into ArgsParser.
```

### Other information

NA

### CODE REVIEW

File: src/main/java/reposense/parser/ArgsParser.java [hunk 1]
```diff
@@ -420,8 +424,13 @@ private static void addAnalysisDatesToBuilder(CliArguments.Builder builder, Name
...
 
-        TimeUtil.verifySinceDateIsValid(sinceDate, currentDate);
-        TimeUtil.verifyDatesRangeIsCorrect(sinceDate, untilDate);
+        if (sinceDate.compareTo(untilDate) > 0) {
+            throw new ParseException(MESSAGE_SINCE_DATE_LATER_THAN_UNTIL_DATE);
+        }
+
+        if (sinceDate.compareTo(currentDate) > 0) {
+            throw new ParseException(MESSAGE_SINCE_DATE_LATER_THAN_TODAY_DATE);
+        }
```
Comment (MarcusTXK) [inline src/main/java/reposense/parser/ArgsParser.java:433]: Could you add some test coverage for the two new if blocks added?

**Review by MarcusTXK — CHANGES_REQUESTED** [review]
Great job, just a small fix and we can merge it in.

File: src/main/java/reposense/parser/ArgsParser.java (hunk 1 above)
Comment (nseah21) [reply src/main/java/reposense/parser/ArgsParser.java]: Hi @MarcusTXK, I've added the tests for ArgParser. The PR is ready for review.

**Review by MarcusTXK — APPROVED** [review]
LGTM, nice job with the refactoring.

**Review by ckcherry23 — APPROVED** [review]
LGTM!

**Pull request merged**